samples/extensions/sparse_image/sparse_image.cpp

```diff
@@ -0,0 +1,1430 @@
+			this->next_stage = SPARSE_IMAGE_COMPARE_MIPS_TABLE_STAGE;
+			break;
+
+		case SPARSE_IMAGE_COMPARE_MIPS_TABLE_STAGE:
+			compare_mips_table();
+			if (virtual_texture.update_required)
+			{
+				this->next_stage = SPARSE_IMAGE_PROCESS_TEXTURE_BLOCKS_STAGE;
+			}
+			else
+			{
+				this->next_stage = SPARSE_IMAGE_CALCULATE_MIPS_TABLE_STAGE;
+			}
+			frame_counter = 0U;
+			break;
+
+		case SPARSE_IMAGE_PROCESS_TEXTURE_BLOCKS_STAGE:
+			process_texture_blocks();
+			this->next_stage = SPARSE_IMAGE_UPDATE_AND_GENERATE_STAGE;
+			break;
+
+		case SPARSE_IMAGE_UPDATE_AND_GENERATE_STAGE:
+			update_and_generate();
+			free_unused_memory();
+
+			if (frame_counter > 10U)
+			{
+				this->next_stage = SPARSE_IMAGE_CALCULATE_MIPS_TABLE_STAGE;
+			}
+			else if (!transfer_finished)
+			{
+				this->next_stage = SPARSE_IMAGE_PROCESS_TEXTURE_BLOCKS_STAGE;
+			}
+			else
+			{
+				virtual_texture.update_required = false;
+				this->next_stage                = SPARSE_IMAGE_CALCULATE_MIPS_TABLE_STAGE;
+			}
+			break;
+
+		default:
+			break;
+	}
+}
+
+void SparseImage::render(float delta_time)
+{
+	if (!prepared)
+	{
+		return;
+	}
+	if (camera.updated)
+	{
+		update_mvp();
+	}
+	if (color_highlight != color_highlight_mem)
+	{
+		update_frag_settings();
+		color_highlight_mem = color_highlight;
+	}
+
+	process_stage(next_stage);
+
+	ApiVulkanSample::prepare_frame();
+	submit_info.commandBufferCount = 1U;
+	submit_info.pCommandBuffers    = &draw_cmd_buffers[current_buffer];
+	VK_CHECK(vkQueueSubmit(queue, 1U, &submit_info, VK_NULL_HANDLE));
+	ApiVulkanSample::submit_frame();
+}
+
+std::unique_ptr<vkb::VulkanSample> create_sparse_image()
+{
+	return std::make_unique<SparseImage>();
+}
+
+/**
+ * 	@brief Generate the mesh and calculate required mip level for each texture block
+ */
+void SparseImage::calculate_mips_table(glm::mat4 mvp_transform, uint32_t numVerticalBlocks, uint32_t numHorizontalBlocks, std::vector<std::vector<MipBlock>> &mipTable)
+{
+	CalculateMipLevelData meshData(mvp_transform, VkExtent2D({static_cast<uint32_t>(virtual_texture.width), static_cast<uint32_t>(virtual_texture.height)}), VkExtent2D({static_cast<uint32_t>(width), static_cast<uint32_t>(height)}), on_screen_num_vertical_blocks, on_screen_num_horizontal_blocks, virtual_texture.mip_levels);
+
+	meshData.calculate_mesh_coordinates();
+	meshData.calculate_mip_levels();
+
+	mipTable = meshData.mip_table;
+}
+
+SparseImage::CalculateMipLevelData::CalculateMipLevelData(glm::mat4 mvp_transform, VkExtent2D texture_base_dim, VkExtent2D screen_base_dim, uint32_t vertical_num_blocks, uint32_t horizontal_num_blocks, uint8_t mip_levels) :
+    mvp_transform(mvp_transform), texture_base_dim(texture_base_dim), screen_base_dim(screen_base_dim), mesh{0}, vertical_num_blocks(vertical_num_blocks), horizontal_num_blocks(horizontal_num_blocks), mip_levels(mip_levels)
```

Comment:
I would order the member initializer list as they are ordered in the class definition.